File: week7/pureabstraction.cpp
#include<iostream>
using namespace std;

class Shapes
{
    private:
    virtual float area()=0;
};

class Triangle: public Shapes
{
    public:
    float b,h;
    Triangle(float base,float height)
    {
        b=base;
        h=height;
    }
    float area()
    {
        return 0.5*b*h;
    }
};

class Circle: public Shapes
{
    public:
    float r;
    Circle(float radius)
    {
        r=radius;
    }
    float area()
    {
        return 3.14*r*r;
    }
};

int main()
{
    Triangle tri(3,45);
    Circle cir(1.5);
    cout<<"Area of the Triangle : "<<tri.area()<<endl;
    cout<<"Area of the Circle: "<<cir.area()<<endl;
    return 0;
}

File: week7/pureabstraction.java
interface Shape
{
    float area();
    void dimensions();
}

class Rectangle implements Shape
{
    float a,b;
    Rectangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return a*b;
    }
    public void dimensions()
    {
        System.out.println("Dimensions : "+a+" X "+b);
    }
}

class Triangle implements Shape
{
    float a,b;
    Triangle(float base,float height)
    {
        a=base;
        b=height;
    }
    public float area()
    {
        return (float)0.5*a*b;
    }
    public void dimensions()
    {
        System.out.println("Dimensions : "+a+" X "+b);
    }
}
public class PureAbstraction 
{
    public static void main(String[] args) 
    {
        Rectangle obj=new Rectangle(7,18);
        Triangle obj1=new Triangle(17,88);
        System.out.println("Area of Rectangle : "+obj.area());obj.dimensions();
        System.out.println("Area of Triangle : "+obj1.area());obj1.dimensions();
    }
}
